vmm/src/cpu.rs

```diff
@@ -1070,7 +1112,35 @@ impl CpuManager {
                             return;
                         }
 
+                    #[cfg(not(feature = "kvm"))]
                     extern "C" fn handle_signal(_: i32, _: *mut siginfo_t, _: *mut c_void) {}
+                    #[cfg(feature = "kvm")]
+                    extern "C" fn handle_signal(_: i32, _: *mut siginfo_t, _: *mut c_void) {
+                        // We do not need a self-pipe for safe UNIX signal handling here as in this
+                        // signal handler, we only expect the same signal over and over again. While
+                        // different signals can interrupt a signal being handled, the same signal
+                        // again can't by default. Therefore, this is safe.
+
+                        // This lock prevents accessing thread locals when a signal is received
+                        // in the teardown phase of the Rust standard library. Otherwise, we would
+                        // panic.
+                        //
+                        // Masking signals would be a nicer approach but this is the pragmatic
+                        // solution.
+                        //
+                        // We don't have lock contention in normal operation. When the writer
+                        // sets the bool to true, the lock is only held for a couple of µs.
+                        let lock = IS_IN_SHUTDOWN.read().unwrap();
+                        if *lock {
+                            return;
+                        }
+
+                        let kvm_run = KVM_RUN.get();
+                        // SAFETY: the mapping is valid
+                        let kvm_run = unsafe {
+                            kvm_run.as_mut().expect("kvm_run should have been mapped as part of vCPU setup") };
```

Comment:
Nit: Does this need to be a binding?
```suggestion
unsafe {
                            kvm_run.as_mut().expect("kvm_run should have been mapped as part of vCPU setup") 
};
```

Reply:
I personally prefer more lines and bindings over "compressed statements". It is mostly a personal preference

Reply:
I just meant that it is very uncommon to bind `()` to a variable (and one that is not used again at that). Am I missing something here? Why doesn't clippy complain?

Reply:
Ah, I see now that I misread the block. The variable is indeed not `()` and used on the next line. All good!

Reply:
> I just meant that it is very uncommon to bind () to a variabl

Okay, no I'm not sure what you are talking about 😁 
The binding is of type `let kvm_run: &mut kvm_bindings::kvm_run = ...`

Reply:
Yeah I also [realized this](https://github.com/cyberus-technology/cloud-hypervisor/pull/21#discussion_r2351084486). I guess we responded around the same time. Anyway it is all good :+1: